client/src/main/java/org/eclipse/hono/client/HonoClient.java

```diff
@@ -338,6 +338,49 @@
     Future<MessageConsumer> createEventConsumer(String tenantId, BiConsumer<ProtonDelivery, Message> eventConsumer,
             Handler<Void> closeHandler);
 
+    Future<MessageConsumer> createCommandResponseConsumer(String tenantId, String replyId, Consumer<Message> commandResponseConsumer,
+            Handler<Void> closeHandler);
+
+    Future<MessageConsumer> createCommandResponseConsumer(String tenantId, String replyId,
+            BiConsumer<ProtonDelivery, Message> commandResponseConsumer,
+            Handler<Void> closeHandler);
+
+    /**
+     * Creates a client for consuming async command responses from Hono's north bound <em>Command API</em>.
+     * <p>
+     * The command responses passed in to the responses consumer will be settled automatically if the consumer does not
+     * throw an exception and does not manually handle the message disposition using the passed in delivery.
+     *
+     * @param tenantId The tenant to consume command responses for.
+     * @param replyId The replyId of commands to consume command responses for.
+     * @param asyncCommandResponseConsumer The handler to invoke with every command response received.
```

Comment:
IMHO simply calling this *consumer* should do

Reply:
done